pkg/transform/kubernetes/kubernetes.go

```diff
@@ -38,6 +38,13 @@ func createServices(app *spec.App) []runtime.Object {
 		if len(svc.Spec.Selector) == 0 {
 			svc.Spec.Selector = app.Labels
 		}
+		if s.Name == "" {
```

Comment:
I have fixed this by adding sanitfy functions in encoding, which populates the missing parts of the spec for now. We can add more to those functions later. @kadel PTAL